### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by arturjordao** [artur-deluca/landscapeviz#3]
Dear Artur de Luca,

Congrats on this implementation, it is very simple and useful to reproduce the loss landscape visualization. 

By running some experiments, I have observed that model.clone becomes the process of generating the loss landscape slower across iterations. By removing the tf.keras.models.clone_model(model) and adding the old weights, at the end of the function, speeds-up roughly 2x the time for producing the landscape and memory consumption was reduced significantly. In addition, by replacing the weights we can remove model.compile and del. Below a preview of the code. 

def obj_fn(model, data, solution):
	w = model.get_weights()
	model.set_weights(solution)
	value = model.evaluate(data[0], data[1], verbose=0)
	model.set_weights(w) #Assings the old weights
	return value

I hope these improvements make your code even better.
Best Regards, 
Artur Jordão.

**Comment by artur-deluca** [comment]
Hey Artur,

Thank you for the feedback! 
I'll look more into it when I have some spare time. 
Feel free to send any PR you judge helpful and I promise to take a look at it, as time permits.

Best,
Artur

**Comment by artur-deluca** [comment]
Hey @arturjordao,

I've taken a look at it. 
To be honest I don't know exactly why have I designed like that.
I'm positive I had some reason but after re-checking it, that old structure doesn't make so much sense.
Thanks for the heads up! :smiley: 

Commit [b1f8f34](https://github.com/artur-deluca/landscapeviz/commit/b1f8f34f104f80b1dafc9b0b6215a2b59abe1d0b)
